tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -654,19 +713,623 @@ TEST_P(RDFVary, VariedHistosMustHaveNoDirectory)
+                .Define("x",
+                        [] {
+                           return ROOT::RVecI{1, 2, 3};
+                        })
+                .Vary(
+                   "x",
+                   [] {
+                      return ROOT::RVec<ROOT::RVecI>{{}, {4, 5}};
+                   },
+                   {}, 2);
+   auto h1 = df.Profile1D<ROOT::RVecI, ROOT::RVecI>({"", "", 100, 0, 100, 0, 100}, "x", "x");
+   auto h1s = VariationsFor(h1);
+   auto h2 = df.Profile2D<ROOT::RVecI, ROOT::RVecI, ROOT::RVecI>({"", "", 100, 0, 100, 100, 0, 100}, "x", "x", "x");
+   auto h2s = VariationsFor(h2);
+
+   EXPECT_DOUBLE_EQ(h1->GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h2->GetMean(), 2.);
+
+   EXPECT_DOUBLE_EQ(h1s["nominal"].GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h2s["nominal"].GetMean(), 2.);
+
+   EXPECT_DOUBLE_EQ(h1s["x:0"].GetMean(), 0.);
+   EXPECT_DOUBLE_EQ(h2s["x:0"].GetMean(), 0.);
+
+   EXPECT_DOUBLE_EQ(h1s["x:1"].GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(h2s["x:1"].GetMean(), 4.5);
+}
+
+TEST(RDFVary, VaryReduce)
+{
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Reduce([](int x, int y) { return x + y; }, "x", 0);
+   auto hs = VariationsFor(h);
+   EXPECT_EQ(*h, 45);
+   EXPECT_EQ(hs["nominal"], 45);
+   EXPECT_EQ(hs["x:0"], 35);
+   EXPECT_EQ(hs["x:1"], 55);
+}
+
+TEST_P(RDFVary, VaryReport)
+{
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Filter([](int x) { return x > 5; }, {"x"}, "before")
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Filter([](int x) { return x > 7; }, {"x"}, "after")
+               .Report();
+   auto &report = *h;
+
+   EXPECT_EQ(report["before"].GetAll(), 10);
+   EXPECT_FLOAT_EQ(report["before"].GetEff(), 40.);
+   EXPECT_EQ(report["before"].GetPass(), 4);
+   EXPECT_EQ(report["after"].GetAll(), 4);
+   EXPECT_FLOAT_EQ(report["after"].GetEff(), 50.);
+   EXPECT_EQ(report["after"].GetPass(), 2);
+   // Varying Reports is not implemented yet, tracked by https://github.com/root-project/root/issues/10551
+   EXPECT_THROW(VariationsFor(h), std::logic_error);
+}
+
+TEST_P(RDFVary, VaryStats)
+{
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Stats<int>("x");
+   EXPECT_THROW(VariationsFor(h), std::runtime_error); // FIXME not (yet) implemented
```

Comment:
I didn't expect to manage this one, so I left it as a runtime error. I am now adapting the test to verify the expected behaviour.

Reply:
you added a test with the PR that fixes Vary+Stats. do we still need this test case?